crates/ruff_linter/src/checkers/ast/annotation.rs

```diff
@@ -59,7 +63,7 @@ impl AnnotationContext {
         // If `__future__` annotations are enabled or it's a stub file,
         // then annotations are never evaluated at runtime,
         // so we can treat them as typing-only.
-        if semantic.future_annotations_or_stub() {
+        if semantic.future_annotations_or_stub() || version.defers_annotations() {
```

Comment:
If there are no situations where we want to treat PEP-563 differently from PEP-649, should we just have the `semantic.future_annotations_or_stub()` method itself always return `true` if the Python version is 3.14 or higher? We could rewrite `SemanticModelFlags::new()` so that it takes a `PythonVersion` argument as well as a `Path` argument, and always sets the `FUTURE_ANNOTATIONS` flag if the Python version is >= 3.14?

https://github.com/astral-sh/ruff/blob/b7ce694162bcd82592926afbbbf918ed61c49bad/crates/ruff_python_semantic/src/model.rs#L2574-L2582

Reply:
I audited existing uses of `future_annotations_or_stub()` and I can't see any uses where it would be problematic if we had the method return `true` for both PEP-563 and PEP-649 being activated